src/services/projects.js

```diff
@@ -448,39 +469,65 @@ module.exports = class ProjectsHelper {
 							})
 						}
 					}
-
 					if (taskEntityType.has_entities) {
 						let checkEntities = utils.checkEntities(taskEntityType, fieldData)
+
 						if (!checkEntities.status) {
+							// process.exit()
+							// let checkNestedEntities = utils.checkEntities(taskEntityType, fieldData)
 							throw responses.failureResponse({
 								message: checkEntities.message,
 								statusCode: httpStatusCode.bad_request,
 								responseCode: 'CLIENT_ERROR',
-								error: utils.errorObject(common.TASKS, entityType.value),
+								error: utils.errorObject(common.TASKS, taskEntityType.value),
 							})
 						}
 					}
 
 					if (taskEntityType.validations.regex) {
-						let checkRegex = utils.checkRegexPattarn(taskEntityType, fieldData)
+						let checkRegex = utils.checkRegexPattern(taskEntityType, fieldData)
 						if (checkRegex) {
 							throw responses.failureResponse({
-								message: `Special characters not allowed in ${entityType.value}`,
+								message: `Special characters not allowed in ${taskEntityType.value}`,
 								statusCode: httpStatusCode.bad_request,
 								responseCode: 'CLIENT_ERROR',
-								error: utils.errorObject(common.TASKS, entityType.value),
+								error: utils.errorObject(common.TASKS, taskEntityType.value),
 							})
 						}
 					}
 				})
 				// TODO: Get file types from products teams and add validation for them
-				if (task.allow_evidences == common.TRUE && task.evidence_details.file_types.length < 1) {
-					throw responses.failureResponse({
-						message: 'FILE_TYPE_NOT_SELECTED',
-						statusCode: httpStatusCode.bad_request,
-						responseCode: 'CLIENT_ERROR',
-						error: utils.errorObject(common.BODY, common.FILE_TYPE),
-					})
+				// entity model mapping add one more key. file types . Ask Nivedhitha accepted file types
+				// dont check file_types.length , change it from array to string. in API - docs also
+				if (task.allow_evidences == common.TRUE) {
+					if (task.evidence_details.file_types.length < 1) {
+						throw responses.failureResponse({
+							message: 'FILE_TYPE_NOT_SELECTED',
+							statusCode: httpStatusCode.bad_request,
+							responseCode: 'CLIENT_ERROR',
+							error: utils.errorObject(common.BODY, common.FILE_TYPE),
+						})
+					}
+
+					let allowed_fileTypes_entitiesList = allowed_fileTypes_entities.reduce((acc, taskEntityType) => {
+						if (taskEntityType.value === common.TASK_ALLOWED_FILE_TYPES) {
```

Comment:
@adithyadinesh0412 instead of can we pick all entities values and then check that value exists in that array. Then no need to go for a difference